lib/dry/system/component.rb

```diff
@@ -1,49 +1,95 @@
-require 'inflecto'
+require 'concurrent/map'
+
+require 'dry-equalizer'
+require 'dry/system/loader'
 
 module Dry
   module System
     class Component
-      attr_reader :loader
-      attr_reader :identifier, :path, :file
+      include Dry::Equalizer(:identifier, :path)
+
+      PATH_SEPARATOR = '/'.freeze
+      WORD_REGEX = /\w+/.freeze
+
+      attr_reader :identifier, :path, :file, :options, :loader
+
+      def self.new(*args)
+        cache.fetch_or_store(args.hash) do
+          name, options = args
+          ns, sep = options.values_at(:namespace, :separator).map(&:to_s)
```

Comment:
I'm guessing not, since this is an "internal" class. I was just doing some playing around on the console and discovering the API as I went, which is why I hit these errors (I also tried ` Dry::System::Component.new("foo", {})`, which got me an `InvalidNamespaceError`, since the separator in that case becomes `nil.to_s`, a blank string, which conflicts with a `nil.to_s` namespace).